src/test/java/org/mifos/integrationtest/cucumber/stepdef/BatchApiStepDef.java

```diff
@@ -602,16 +605,19 @@ public void iCallTheBatchSummaryAPIForSubBatchSummaryWithExpectedStatusOf(int ex
 
     @Then("I am able to parse sub batch summary response")
     public void iAmAbleToParseSubBatchSummaryResponse() {
-        scenarioScopeState.batchAndSubBatchSummaryResponse = null;
-        assertThat(scenarioScopeState.response).isNotNull();
-        assertThat(scenarioScopeState.response).isNotEmpty();
-        try {
-            scenarioScopeState.batchAndSubBatchSummaryResponse = objectMapper.readValue(scenarioScopeState.response,
-                    BatchAndSubBatchSummaryResponse.class);
-        } catch (Exception e) {
-            logger.error("Error parsing the batch summary response", e);
-        }
-        assertThat(scenarioScopeState.batchAndSubBatchSummaryResponse).isNotNull();
+        await().atMost(awaitMost, SECONDS).pollDelay(pollDelay, SECONDS).pollInterval(pollInterval, SECONDS).untilAsserted(() -> {
```

Comment:
here again, Awaitly is not required here, there are no external request in this block